.github/workflows/build-and-test.yml

```diff
@@ -354,9 +354,12 @@ jobs:
     runs-on: ubuntu-latest
     needs: [lint, unittest, integration-tests, build-package, windows-msi]
     if: (github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v')) && github.repository == 'open-telemetry/opentelemetry-collector-contrib'
+    permissions:
```

Comment:
I believe tracegen should be published as part of a different workflow, that is executed only with the "push" and "release" events. I believe the Docker actions are already doing the right things, but on push, it should update the latest tag, and on release, it should use the version number.

Reply:
fixed.